compiler-plugin/src/main/java/io/ballerina/stdlib/sql/compiler/analyzer/MethodAnalyzer.java

```diff
@@ -98,7 +98,15 @@ public void perform(SyntaxNodeAnalysisContext ctx) {
         }
         TypeSymbol argumentTypeSymbol = ((TypeSymbol) typeDescriptionArgument.get());
         TypeDescKind argTypeKind = argumentTypeSymbol.typeKind();
-        DiagnosticInfo diagnosticsForInvalidTypes = Utils.addDiagnosticsForInvalidTypes(objectName, argTypeKind);
+        String typeName = null;
+        if (argTypeKind == TypeDescKind.TYPE_REFERENCE) {
+            TypeSymbol typeDescriptor = ((TypeReferenceTypeSymbol) argumentTypeSymbol).typeDescriptor();
+            argTypeKind = typeDescriptor.typeKind();
+            typeName = argumentTypeSymbol.signature();
+        }
```

Comment:
Let's move this into `addDiagnosticsForInvalidTypes()` function, which takes the argument symbol and processes it.